File: frontend/src/app/models/ItensServico.ts
import { Funcionario } from "./Funcionario";
import { OrdemServico } from "./Os";
import { Servico } from "./Servico";

export interface ItensServico {
    id: number;
    horarioInicio: string; // Usando string para representar horários no formato ISO ou 'HH:mm:ss'
    horarioFim: string;
    quantidade: number;
    precoTotal: number;
    idFuncionario: Funcionario; // Relacionamento com Funcionario
    idServico: Servico; // Relacionamento com Servico
    numeroOs: OrdemServico; // Relacionamento com OrdemServico
  }
  

File: backend/src/main/java/com/sistemaOficina/backend/entity/ItensServico.java
package com.sistemaOficina.backend.entity;

import lombok.AllArgsConstructor;
import lombok.Data;

import java.time.LocalTime;

@Data
@AllArgsConstructor
public class ItensServico {
    private int id;
    private LocalTime horarioInicio;
    private LocalTime horarioFim;
    private int quantidade;
    private double precoTotal;
    private Funcionario idFuncionario;
    private Servico idServico;   
    private OrdemServico numeroOs;      
}
